nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -30,6 +46,30 @@ in
         description = "Extra flags passed to the GlusterFS daemon";
         default = [];
       };
+
+      enableTLS = mkOption {
+        type = types.bool;
+        description = "Secure communication via TLS";
+        default = false;
+      };
+
+      tlsKey = mkOption {
+        default = null;
+        type = types.nullOr types.path;
+        description = "GlusterFS path of TLS private key.";
+      };
+
+      tlsPem = mkOption {
+        default = null;
+        type = types.nullOr types.path;
+        description = "GlusterFS path of TLS certificate.";
+      };
+
+      caCert = mkOption {
+        default = null;
```

Comment:
There should be some `assert`s here that ensure that when `enableTLS` is true, `tlsKey` and `tlsKey` and `tlsPem` are not null.

Better yet, make a `tlsSettings` option of type `submodule` that can be either `null` (for TLS being disabled), or a `{}` with the 3 options inside, each non-nullable.

An example of how to do that can be found here:

https://github.com/nh2/nixops-gluster-example/blob/8a859f0a702efd38bef9a202e1b031f1db6a44d3/modules/nh2-glusterfs-server.nix#L128-L160